scanbd fails to cross compile

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build .#pkgsCross.aarch64-multiplatform.scanbd`

### Build log
```
       > /nix/store/5yrngpzwj81l75rwb2abl082jn8awj7v-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: config.o: in function `make_script_path_abs':
       > /build/1.5.1/src/scanbd/config.c:151: undefined reference to `rpl_malloc'
       > /nix/store/5yrngpzwj81l75rwb2abl082jn8awj7v-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: sane.o: in function `sane_find_matching_options':
       > /build/1.5.1/src/scanbd/sane.c:553: undefined reference to `rpl_malloc'
       > /nix/store/5yrngpzwj81l75rwb2abl082jn8awj7v-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: /build/1.5.1/src/scanbd/sane.c:568: undefined reference to `rpl_malloc'
       > collect2: error: ld returned 1 exit status
       > make[1]: *** [Makefile:479: scanbd] Error 1
       > make[1]: Leaving directory '/build/1.5.1/src/scanbd'
       > make: *** [Makefile:425: all-recursive] Error 1
```

### Additional context
This is the same problem as addressed here for cifs-utils: https://github.com/NixOS/nixpkgs/pull/106761

--- a/pkgs/tools/graphics/scanbd/default.nix
+++ b/pkgs/tools/graphics/scanbd/default.nix
@@ -18,6 +18,10 @@ stdenv.mkDerivation rec {
     "--enable-udev"
     "--with-scanbdconfdir=/etc/scanbd"
     "--with-systemdsystemunitdir=$out/lib/systemd/system"
+  ] ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+    # AC_FUNC_MALLOC is broken on cross builds.
+    "ac_cv_func_malloc_0_nonnull=yes"
+    "ac_cv_func_realloc_0_nonnull=yes"
   ];
 
   enableParallelBuilding = true;

